uwblueprint/community-fridge-kw
# feat: Donor API routes

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Create-Donor-Endpoints-417510ecbbbb4b75b7a58ff594393aef)



## Implementation description

The service for the donor model was created along with API endpoints.

The following endpoints were created:

* donors.get("/")
* donors.get("/:donorId")
* donors.put("/:donorId")
* donors.delete("/:donorID")



## Steps to test
* Open Postman
* Test the users.post("/") method where the role is Donor
* Test the get("/") method:
  * Check if the donor is visible
* Test the get("/:donorID") method:
  * Check if the donor returned has same user_id as what was passed as donor_id
* Test the put("/donorID") method:
  * Check if the donor is updated with new info
* Test the delete("/donorID") method:
  * Check if the donor was deleted


## What should reviewers focus on?
* Edge cases for post, put and delete APIs


## Checklist
